.github/workflows/doc-upload.yml

```diff
@@ -0,0 +1,26 @@
+name: Documentation Uploading
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  PR-check:
+    uses: ./.github/workflows/PR-check.yml
+
+  doc_upload:
+    runs-on: ubuntu-22.04
+    needs: PR-check
+    steps:
+    - name: Get docs
+      uses: actions/download-artifact@v4
+      with:
+        name: docs
+    - name: do something
```

Comment:
@arodebaugh can you take a look at this [page](https://docs.github.com/en/pages/getting-started-with-github-pages/using-custom-workflows-with-github-pages) and figure out how to deploy the docs artifact to your pages, then add the needed code here?